pkg/cloudprovider/providers/vsphere/vclib/datacenter.go

```diff
@@ -82,6 +82,33 @@ func (dc *Datacenter) GetVMByUUID(ctx context.Context, vmUUID string) (*VirtualM
 	return &virtualMachine, nil
 }
 
+// GetHostByVMUUID gets the Host object from the given vmUUID
+func (dc *Datacenter) GetHostByVMUUID(ctx context.Context, vmUUID string) (*types.ManagedObjectReference, error) {
+	s := object.NewSearchIndex(dc.Client())
+	vmUUID = strings.ToLower(strings.TrimSpace(vmUUID))
+	svm, err := s.FindByUuid(ctx, dc.Datacenter, vmUUID, true, nil)
+	if err != nil {
+		glog.Errorf("Failed to find VM by UUID. VM UUID: %s, err: %+v", vmUUID, err)
+		return nil, err
```

Comment:
thanks for catching this! But looking at other code in this file, it seems that `glog.Errorf` is using more frequently. @dougm there are 44 places that using `glog.Errof`. What do you think about this or `fmt.Errorf`?

Reply:
There are cases in VCP where `return fmt.Errorf` is used too.  I think its ok to leave as `glog.Errorf` as it can be useful to have the `filename:linenumber` info that it provides.  We should probably have a separate issue to clean up the the logging in VCP.  In this particular case, looks like this code should be calling `Datacenter.GetVMByUUID` instead as I commented below.